osu.Game/Rulesets/Edit/IPositionSnapProvider.cs

```diff
@@ -5,23 +5,27 @@
 
 namespace osu.Game.Rulesets.Edit
 {
+    /// <summary>
+    /// A snap provider which given a proposed position for a hit object, potentially offers a more correct position and time value inferred from the context of the beatmap.
+    /// Provided values are inferred in an isolated context, without consideration of other nearby hit objects.
+    /// </summary>
     public interface IPositionSnapProvider
     {
         /// <summary>
         /// Given a position, find a valid time and position snap.
         /// </summary>
         /// <remarks>
-        /// This call should be equivalent to running <see cref="SnapScreenSpacePositionToValidPosition"/> with any additional logic that can be performed without the time immutability restriction.
+        /// This call should be equivalent to running <see cref="FindSnappedPosition"/> with any additional logic that can be performed without the time immutability restriction.
         /// </remarks>
         /// <param name="screenSpacePosition">The screen-space position to be snapped.</param>
         /// <returns>The time and position post-snapping.</returns>
-        SnapResult SnapScreenSpacePositionToValidTime(Vector2 screenSpacePosition);
+        SnapResult FindSnappedPositionAndTime(Vector2 screenSpacePosition);
 
         /// <summary>
         /// Given a position, find a value position snap, restricting time to its input value.
```

Comment:
While we're here adjusting xmldocs, am I the only one that finds this one illegible? "value position snap" seems to be a typo, but I'd generally reword this to something like

```suggestion
        /// Given a position, find a valid position snap, without changing the time value.
```